--- llvm-project-llvmorg-20.1.8/llvm/include/llvm/ADT/bit.h	2025-07-08 23:06:32.000000000 +0000
+++ -	2025-11-24 00:10:43.673845333 +0000
@@ -29,7 +29,7 @@
 
 #if defined(__linux__) || defined(__GNU__) || defined(__HAIKU__) ||            \
     defined(__Fuchsia__) || defined(__EMSCRIPTEN__) || defined(__NetBSD__) ||  \
-    defined(__OpenBSD__) || defined(__DragonFly__)
+    defined(__OpenBSD__) || defined(__DragonFly__) || defined(__COSMOPOLITAN__)
 #include <endian.h>
 #elif defined(_AIX)
 #include <sys/machine.h>
--- llvm-project-llvmorg-20.1.8/llvm/lib/Support/Unix/Path.inc	2025-07-08 23:06:32.000000000 +0000
+++ -	2025-11-24 00:16:01.897591508 +0000
@@ -50,6 +50,8 @@
 #elif defined(__MVS__)
 #include "llvm/Support/AutoConvert.h"
 #include <sys/ps.h>
+#elif defined(__COSMOPOLITAN__)
+#include <cosmo.h>
 #endif
 
 // Both stdio.h and cstdio are included via different paths and
@@ -107,7 +109,7 @@
 #endif
 
 #if defined(__NetBSD__) || defined(__DragonFly__) || defined(__GNU__) ||       \
-    defined(__MVS__)
+    defined(__MVS__) || defined(__COSMOPOLITAN__)
 #define STATVFS_F_FLAG(vfs) (vfs).f_flag
 #else
 #define STATVFS_F_FLAG(vfs) (vfs).f_flags
@@ -124,7 +126,8 @@
 #if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) ||      \
     defined(__FreeBSD_kernel__) || defined(__linux__) || defined(__CYGWIN__) || \
     defined(__DragonFly__) || defined(_AIX) || defined(__GNU__) ||              \
-    (defined(__sun__) && defined(__svr4__) || defined(__HAIKU__))
+    (defined(__sun__) && defined(__svr4__) || defined(__HAIKU__)) || \
+    defined(__COSMOPOLITAN__)
 static int test_dir(char ret[PATH_MAX], const char *dir, const char *bin) {
   struct stat sb;
   char fullpath[PATH_MAX];
@@ -329,6 +332,8 @@
   char link_path[PATH_MAX];
   if (realpath(DLInfo.dli_fname, link_path))
     return link_path;
+#elif defined(__COSMOPOLITAN__)
+  return GetProgramExecutableName();
 #else
 #error GetMainExecutable is not implemented on this host yet.
 #endif
@@ -550,6 +555,8 @@
   // The file system can have an arbitrary structure on z/OS; must go with the
   // conservative answer.
   return false;
+#elif defined(__COSMOPOLITAN__)
+  return false;
 #else
   return !!(STATVFS_F_FLAG(Vfs) & MNT_LOCAL);
 #endif
--- llvm-project-llvmorg-20.1.8/llvm/lib/Support/Unix/Threading.inc	2025-07-08 23:06:32.000000000 +0000
+++ -	2025-11-24 00:17:30.965483297 +0000
@@ -383,7 +383,7 @@
   }
   return CPU_COUNT(&Enabled);
 }
-#elif (defined(__linux__) && defined(__s390x__)) || defined(_AIX)
+#elif (defined(__linux__) && defined(__s390x__)) || defined(_AIX) || defined(__COSMOPOLITAN__)
 static int computeHostNumPhysicalCores() {
   return sysconf(_SC_NPROCESSORS_ONLN);
 }
--- llvm-project-llvmorg-20.1.8/llvm/lib/Support/Unix/Signals.inc	2025-07-08 23:06:32.000000000 +0000
+++ -	2025-11-24 00:20:35.474179322 +0000
@@ -817,7 +817,7 @@
   OS << "Stack dump without symbol names (ensure you have llvm-symbolizer in "
         "your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point "
         "to it):\n";
-#if HAVE_DLOPEN && !defined(_AIX)
+#if HAVE_DLOPEN && !defined(_AIX) && !defined(__COSMOPOLITAN__)
   int width = 0;
   for (int i = 0; i < depth; ++i) {
     Dl_info dlinfo;
--- llvm-project-llvmorg-20.1.8/llvm/lib/Support/Unix/Path.inc	2025-11-24 00:18:24.072322435 +0000
+++ -	2025-11-24 00:23:37.537441140 +0000
@@ -320,7 +320,7 @@
       return std::string(real_path);
     break; // Found entry, but realpath failed.
   }
-#elif defined(HAVE_DLOPEN)
+#elif defined(HAVE_DLOPEN) && !defined(__COSMOPOLITAN__)
   // Use dladdr to get executable path if available.
   Dl_info DLInfo;
   int err = dladdr(MainAddr, &DLInfo);
--- llvm-project-llvmorg-20.1.8/llvm/tools/dsymutil/DwarfLinkerForBinary.cpp	2025-11-24 01:31:14.542389566 +0000
+++ -	2025-11-24 01:31:29.964541057 +0000
@@ -176,7 +176,7 @@
   std::string Message = FE->message();
   Error E = FE->takeError();
   Error NewE = handleErrors(std::move(E), [&](std::unique_ptr<ECError> EC) {
-    if (EC->convertToErrorCode() != std::errc::no_such_file_or_directory)
+    if (EC->convertToErrorCode() != std::make_error_code(std::errc::no_such_file_or_directory))
       return Error(std::move(EC));
 
     Linker.reportWarning(Message, DMO.getObjectFilename());
