diff -urN ./ninja-1.11.1/src/disk_interface.cc /home/ahgamut/sources/ninja-1.11.1/src/disk_interface.cc
--- ./ninja-1.11.1/src/disk_interface.cc	2022-08-30 19:47:02.000000000 +0000
+++ -    2023-11-09 05:11:13.543909186 +0000
@@ -258,11 +258,10 @@
 FileReader::Status RealDiskInterface::ReadFile(const string& path,
                                                string* contents,
                                                string* err) {
-  switch (::ReadFile(path, contents, err)) {
-  case 0:       return Okay;
-  case -ENOENT: return NotFound;
-  default:      return OtherError;
-  }
+  auto zz = ::ReadFile(path, contents, err);
+  if (zz == 0)       return Okay;
+  if (zz == -ENOENT) return NotFound;
+  return OtherError;
 }
 
 int RealDiskInterface::RemoveFile(const string& path) {
@@ -308,13 +307,10 @@
   }
 #else
   if (remove(path.c_str()) < 0) {
-    switch (errno) {
-      case ENOENT:
+      if(errno == ENOENT)
         return 1;
-      default:
-        Error("remove(%s): %s", path.c_str(), strerror(errno));
-        return -1;
-    }
+      Error("remove(%s): %s", path.c_str(), strerror(errno));
+      return -1;
   }
 #endif
   return 0;
diff -urN ./ninja-1.11.1/superconfigure /home/ahgamut/sources/ninja-1.11.1/superconfigure
--- ./ninja-1.11.1/superconfigure	1970-01-01 00:00:00.000000000 +0000
+++ -    2023-11-09 05:12:10.633713181 +0000
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+mkdir -p build
+cd build
+ninja -t clean || true
+
+cmake -G Ninja ..\
+    -DCMAKE_BUILD_SHARED=OFF\
+    -DCMAKE_BUILD_STATIC=ON\
+    -DCMAKE_INSTALL_PREFIX="$COSMOS"
