--- openssl-openssl-3.3.1/crypto/bio/bio_sock.c	2024-07-24 11:46:04.574188758 +0000
+++ -	2024-07-24 11:46:23.691976275 +0000
@@ -7,8 +7,11 @@
  * https://www.openssl.org/source/license.html
  */
 
+#ifdef __COSMOPOLITAN__
+#define _COSMO_SOURCE
 #include <stdio.h>
 #include <stdlib.h>
+#endif
 #include "bio_local.h"
 #ifndef OPENSSL_NO_SOCK
 # define SOCKET_PROTOCOL IPPROTO_TCP
--- openssl-openssl-3.3.1/providers/implementations/rands/seeding/rand_unix.c	2024-06-04 12:53:04.000000000 +0000
+++ -	2024-08-25 17:09:07.361096840 +0000
@@ -40,6 +40,9 @@
 # include <sys/param.h>
 # include <sys/random.h>
 #endif
+#if defined(__COSMOPOLITAN__)
+# include <sys/random.h>
+#endif
 
 #if (defined(OPENSSL_SYS_UNIX) && !defined(OPENSSL_SYS_VXWORKS)) \
      || defined(__DJGPP__)
@@ -393,7 +396,8 @@
 #  elif (defined(__FreeBSD__) || defined(__NetBSD__)) && defined(KERN_ARND)
     return sysctl_random(buf, buflen);
 #  elif (defined(__DragonFly__)  && __DragonFly_version >= 500700) \
-     || (defined(__NetBSD__) && __NetBSD_Version >= 1000000000)
+     || (defined(__NetBSD__) && __NetBSD_Version >= 1000000000) \
+     || (defined(__COSMOPOLITAN__))
     return getrandom(buf, buflen, 0);
 #  elif defined(__wasi__)
     if (getentropy(buf, buflen) == 0)
