diff -r --unified SDL2-2.28.5-old/configure.ac SDL2-2.28.5/configure.ac
--- SDL2-2.28.5-old/configure.ac	2023-11-02 18:03:38.000000000 +0100
+++ SDL2-2.28.5/configure.ac	2023-12-02 21:51:13.884690520 +0100
@@ -349,7 +349,7 @@
         AC_DEFINE(HAVE_MPROTECT, 1, [ ])
         ],[]),
     )
-    AC_CHECK_FUNCS(malloc calloc realloc free getenv setenv putenv unsetenv bsearch qsort abs bcopy memset memcmp memcpy memmove wcslen wcslcpy wcslcat _wcsdup wcsdup wcsstr wcscmp wcsncmp wcscasecmp _wcsicmp wcsncasecmp _wcsnicmp strlen strlcpy strlcat _strrev _strupr _strlwr index rindex strchr strrchr strstr strtok_r itoa _ltoa _uitoa _ultoa strtod strtol strtoul _i64toa _ui64toa strtoll strtoull atoi atof strcmp strncmp _stricmp strcasecmp _strnicmp strncasecmp strcasestr vsscanf vsnprintf fopen64 fseeko fseeko64 sigaction setjmp nanosleep sysconf sysctlbyname getauxval elf_aux_info poll _Exit)
+    AC_CHECK_FUNCS(malloc calloc realloc free getenv setenv putenv unsetenv bsearch qsort abs bcopy memset memcmp memcpy memmove wcslen wcslcpy wcslcat _wcsdup wcsdup wcsstr wcscmp wcsncmp wcscasecmp _wcsicmp wcsncasecmp _wcsnicmp strlen strlcpy strlcat _strrev _strupr _strlwr index rindex strchr strrchr strstr strtok_r itoa _ltoa _uitoa _ultoa strtod strtol strtoul _i64toa _ui64toa strtoll strtoull atoi atof strcmp strncmp _stricmp strcasecmp _strnicmp strncasecmp strcasestr vsscanf vsnprintf fopen64 fseeko fseeko64 sigaction setjmp nanosleep sysconf getauxval elf_aux_info poll _Exit)
 
     AC_CHECK_LIB(m, pow, [LIBS="$LIBS -lm"; EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lm"])
     AC_CHECK_FUNCS(acos acosf asin asinf atan atanf atan2 atan2f ceil ceilf copysign copysignf cos cosf exp expf fabs fabsf floor floorf trunc truncf fmod fmodf log logf log10 log10f lround lroundf pow powf round roundf scalbn scalbnf sin sinf sqrt sqrtf tan tanf)
@@ -1906,6 +1906,7 @@
                 fi
                 ;;
         esac
+        LIBS="$LIBS -lxcb -lXau"
         AC_PATH_X
         AC_PATH_XTRA
         if test x$have_x = xyes; then
diff -r --unified SDL2-2.28.5-old/src/thread/pthread/SDL_systhread.c SDL2-2.28.5/src/thread/pthread/SDL_systhread.c
--- SDL2-2.28.5-old/src/thread/pthread/SDL_systhread.c	2023-11-02 18:03:38.000000000 +0100
+++ SDL2-2.28.5/src/thread/pthread/SDL_systhread.c	2023-12-02 21:49:06.324710657 +0100
@@ -18,6 +18,7 @@
      misrepresented as being the original software.
   3. This notice may not be removed or altered from any source distribution.
 */
+#include <libc/sysv/consts/sched.h>
 #include "../../SDL_internal.h"
 
 #include "SDL_system.h"
--- SDL2-2.28.5/sdl2.pc.in	2023-11-02 17:03:38.000000000 +0000
+++ -	2023-12-03 06:03:03.576900598 +0000
@@ -10,5 +10,5 @@
 Version: @SDL_VERSION@
 Requires.private: @PKGCONFIG_DEPENDS@
 Conflicts:
-Libs: -L${libdir} @SDL_RLD_FLAGS@ @SDL_LIBS@ @PKGCONFIG_LIBS_PRIV@ @SDL_STATIC_LIBS@
+Libs: -L${libdir}  -lSDL2   -lm  -lX11 -lXext -lxcb -lXau -lpthread
 Cflags: -I${includedir} -I${includedir}/SDL2 @SDL_CFLAGS@
--- SDL2-2.28.5/src/cpuinfo/SDL_cpuinfo.c	2023-11-02 17:03:38.000000000 +0000
+++ -	2025-01-23 11:58:03.767987037 +0000
@@ -41,6 +41,8 @@
 #include "SDL_cpuinfo.h"
 #include "SDL_assert.h"
 
+#undef HAVE_SYSCTLBYNAME
+
 #ifdef HAVE_SYSCONF
 #include <unistd.h>
 #endif
@@ -343,7 +345,7 @@
 #endif
     int hasVectorUnit = 0;
     size_t length = sizeof(hasVectorUnit);
-    int error = sysctl(selectors, 2, &hasVectorUnit, &length, NULL, 0);
+    int error = -1; // sysctl(selectors, 2, &hasVectorUnit, &length, NULL, 0);
     if (0 == error) {
         altivec = (hasVectorUnit != 0);
     }
@@ -1092,7 +1094,7 @@
             Uint64 memsize = 0;
             size_t len = sizeof(memsize);
 
-            if (sysctl(mib, 2, &memsize, &len, NULL, 0) == 0) {
+            if (-1 /* sysctl(mib, 2, &memsize, &len, NULL, 0) */ == 0) {
                 SDL_SystemRAM = (int)(memsize / (1024 * 1024));
             }
         }
