--- tmux-3.3a/input.c	2022-06-09 11:30:50.000000000 +0000
+++ -	2023-11-27 03:43:32.246271476 +0000
@@ -21,7 +21,7 @@
 #include <netinet/in.h>
 
 #include <ctype.h>
-#include <resolv.h>
+// #include <resolv.h>
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
--- tmux-3.3a/tty-keys.c	2022-06-09 11:30:50.000000000 +0000
+++ -	2023-11-27 03:44:25.829028302 +0000
@@ -23,7 +23,7 @@
 
 #include <ctype.h>
 #include <limits.h>
-#include <resolv.h>
+// #include <resolv.h>
 #include <stdlib.h>
 #include <string.h>
 #include <termios.h>
--- tmux-3.3a/tty.c	2022-06-09 11:30:50.000000000 +0000
+++ -	2023-11-27 03:44:37.742644047 +0000
@@ -24,7 +24,7 @@
 #include <curses.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <resolv.h>
+// #include <resolv.h>
 #include <stdlib.h>
 #include <string.h>
 #include <termios.h>
--- tmux-3.3a/compat/base64.c	2022-06-09 11:30:50.000000000 +0000
+++ -	2023-11-27 03:45:09.815703828 +0000
@@ -49,7 +49,7 @@
-#include <arpa/nameser.h>
+// #include <arpa/nameser.h>
 
 #include <ctype.h>
-#include <resolv.h>
+// #include <resolv.h>
 #include <stdio.h>
 
 #include <stdlib.h>
--- tmux-3.3a/osdep-linux.c	2022-06-09 11:30:50.000000000 +0000
+++ -	2023-11-27 03:48:29.165423396 +0000
@@ -75,7 +75,7 @@
 	n = readlink(path, target, MAXPATHLEN);
 	free(path);
 
-	if (n == -1 && ioctl(fd, TIOCGSID, &sid) != -1) {
+	if (n == -1 /* && ioctl(fd, TIOCGSID, &sid) != -1 */) {
 		xasprintf(&path, "/proc/%lld/cwd", (long long) sid);
 		n = readlink(path, target, MAXPATHLEN);
 		free(path);
