--- ddrescue-1.29/configure	2025-01-05 13:02:32.000000000 +0000
+++ -	2025-09-30 12:46:08.651439781 +0000
@@ -15,13 +15,13 @@
 LC_ALL=C
 export LC_ALL
 srcdir=
-prefix=/usr/local
+prefix="$COSMOS"
 exec_prefix='$(prefix)'
 bindir='$(exec_prefix)/bin'
 datarootdir='$(prefix)/share'
 infodir='$(datarootdir)/info'
 mandir='$(datarootdir)/man'
-CXX=g++
+CXX="$CXX"
 CPPFLAGS=
 CXXFLAGS='-Wall -W -O2'
 LDFLAGS=
