diff --git a/Makefile b/Makefile
index a84af6b..95adcd5 100755
--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,12 @@
-CFLAGS      = -Wall -Wextra -std=c99 -pedantic-errors -O2
+CFLAGS      = -Wall -Wextra -std=c99 -pedantic-errors -O2 -I$(COSMOS)/include
 DEBUG_FLAGS = -O0 -g -DBE_DEBUG
 TEST_FLAGS  = $(DEBUG_FLAGS) --coverage -fno-omit-frame-pointer -fsanitize=address -fsanitize=undefined
-LIBS        = -lm
+LIBS        = -static -lm -ldl -lreadline -lncurses -ltinfo
 TARGET      = berry
-CC          = gcc
+CC          = $(COSMOS)/bin/cosmocc
 MKDIR       = mkdir
-LFLAGS      =
-PREFIX      = /usr/local
+LFLAGS      = -L$COSMOS/lib
+PREFIX      = $(COSMOS)
 BINDIR      = $(PREFIX)/bin
 
 INCPATH     = src default
