--- src/include/postmaster/syslogger.h	2023-11-30 18:03:08.094066788 +0000
+++ -	2023-11-30 18:03:20.414395138 +0000
@@ -29,7 +29,7 @@
  * we still might find it in the len and/or pid bytes unless we're careful.
  */
 
-#ifdef PIPE_BUF
+#if 0 && PIPE_BUF
 /* Are there any systems with PIPE_BUF > 64K?  Unlikely, but ... */
 #if PIPE_BUF > 65536
 #define PIPE_CHUNK_SIZE  65536
--- src/include/pg_config.h.in	2023-11-06 22:06:45.000000000 +0000
+++ -	2023-11-30 18:06:02.450768656 +0000
@@ -305,7 +305,7 @@
 #undef HAVE_INT_TIMEZONE
 
 /* Define to 1 if you have support for IPv6. */
-#undef HAVE_IPV6
+#undef HAVE_IPV6_PLS_NO
 
 /* Define to 1 if __builtin_constant_p(x) implies "i"(x) acceptance. */
 #undef HAVE_I_CONSTRAINT__BUILTIN_CONSTANT_P
