diff -r --unified zsh-5.9a/Src/Modules/cap.mdd zsh-5.9/Src/Modules/cap.mdd
--- zsh-5.9a/Src/Modules/cap.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/cap.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/cap
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:cap b:getcap b:setcap"
diff -r --unified zsh-5.9a/Src/Modules/clone.mdd zsh-5.9/Src/Modules/clone.mdd
--- zsh-5.9a/Src/Modules/clone.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/clone.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/clone
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:clone"
diff -r --unified zsh-5.9a/Src/Modules/curses.mdd zsh-5.9/Src/Modules/curses.mdd
--- zsh-5.9a/Src/Modules/curses.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/curses.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/curses
-link='if test "x$ac_cv_func_initscr" = xyes && test "x$zsh_cv_path_curses_header" != x; then echo dynamic; else echo no; fi'
+link=no
 load=no
 
 autofeatures="b:zcurses"
diff -r --unified zsh-5.9a/Src/Modules/datetime.mdd zsh-5.9/Src/Modules/datetime.mdd
--- zsh-5.9a/Src/Modules/datetime.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/datetime.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,6 +1,6 @@
 name=zsh/datetime
 
-link=either
+link=static
 load=no
 
 functions='Functions/Calendar/*'
diff -r --unified zsh-5.9a/Src/Modules/db_gdbm.mdd zsh-5.9/Src/Modules/db_gdbm.mdd
--- zsh-5.9a/Src/Modules/db_gdbm.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/db_gdbm.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,10 +1,5 @@
 name=zsh/db/gdbm
-link='if test "x$ac_cv_lib_gdbm_gdbm_open" = xyes && test "x$ac_cv_header_gdbm_h" = xyes; then
-  echo dynamic
-else
-  echo no
-fi
-'
+link=static
 load=no
 
 autofeatures="b:ztie b:zuntie b:zgdbmpath p:zgdbm_tied"
diff -r --unified zsh-5.9a/Src/Modules/example.mdd zsh-5.9/Src/Modules/example.mdd
--- zsh-5.9a/Src/Modules/example.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/example.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/example
-link=dynamic
+link=no
 load=no
 
 autofeatures="b:example C:ex c:len p:exint p:exstr p:exarr f:sum f:length"
diff -r --unified zsh-5.9a/Src/Modules/files.mdd zsh-5.9/Src/Modules/files.mdd
--- zsh-5.9a/Src/Modules/files.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/files.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/files
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:chgrp b:chown b:ln b:mkdir b:mv b:rm b:rmdir b:sync b:zf_chgrp b:zf_chown b:zf_ln b:zf_mkdir b:zf_mv b:zf_rm b:zf_rmdir b:zf_sync"
diff -r --unified zsh-5.9a/Src/Modules/mapfile.mdd zsh-5.9/Src/Modules/mapfile.mdd
--- zsh-5.9a/Src/Modules/mapfile.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/mapfile.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/mapfile
-link=dynamic
+link=static
 load=no
 
 autofeatures="p:mapfile"
diff -r --unified zsh-5.9a/Src/Modules/mathfunc.mdd zsh-5.9/Src/Modules/mathfunc.mdd
--- zsh-5.9a/Src/Modules/mathfunc.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/mathfunc.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/mathfunc
-link=dynamic
+link=static
 load=no
 
 objects="mathfunc.o"
diff -r --unified zsh-5.9a/Src/Modules/nearcolor.mdd zsh-5.9/Src/Modules/nearcolor.mdd
--- zsh-5.9a/Src/Modules/nearcolor.mdd	2019-02-01 00:37:34.000000000 +0000
+++ zsh-5.9/Src/Modules/nearcolor.mdd	2023-12-05 00:00:29.933471423 +0000
@@ -1,5 +1,5 @@
 name=zsh/nearcolor
-link=dynamic
+link=static
 load=no
 
 objects="nearcolor.o"
diff -r --unified zsh-5.9a/Src/Modules/param_private.mdd zsh-5.9/Src/Modules/param_private.mdd
--- zsh-5.9a/Src/Modules/param_private.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/param_private.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/param/private
-link=dynamic
+link=static
 load=yes
 
 autofeatures="b:private"
diff -r --unified zsh-5.9a/Src/Modules/parameter.mdd zsh-5.9/Src/Modules/parameter.mdd
--- zsh-5.9a/Src/Modules/parameter.mdd	2018-11-13 08:04:43.000000000 +0000
+++ zsh-5.9/Src/Modules/parameter.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/parameter
-link=either
+link=static
 load=yes
 
 autofeatures="p:parameters p:commands p:functions p:dis_functions p:functions_source p:dis_functions_source p:funcfiletrace p:funcsourcetrace p:funcstack p:functrace p:builtins p:dis_builtins p:reswords p:dis_reswords p:patchars p:dis_patchars p:options p:modules p:dirstack p:history p:historywords p:jobtexts p:jobdirs p:jobstates p:nameddirs p:userdirs p:usergroups p:aliases p:dis_aliases p:galiases p:dis_galiases p:saliases p:dis_saliases"
diff -r --unified zsh-5.9a/Src/Modules/socket.mdd zsh-5.9/Src/Modules/socket.mdd
--- zsh-5.9a/Src/Modules/socket.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/socket.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/net/socket
-link=dynamic
+link=static
 load=no
 
 objects="socket.o"
diff -r --unified zsh-5.9a/Src/Modules/stat.mdd zsh-5.9/Src/Modules/stat.mdd
--- zsh-5.9a/Src/Modules/stat.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/stat.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/stat
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:stat b:zstat"
diff -r --unified zsh-5.9a/Src/Modules/system.mdd zsh-5.9/Src/Modules/system.mdd
--- zsh-5.9a/Src/Modules/system.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/system.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/system
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:sysread b:syswrite b:sysopen b:sysseek b:syserror p:errnos f:systell"
diff -r --unified zsh-5.9a/Src/Modules/tcp.h zsh-5.9/Src/Modules/tcp.h
--- zsh-5.9a/Src/Modules/tcp.h	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/tcp.h	2023-12-05 00:00:29.937471863 +0000
@@ -57,7 +57,7 @@
  * not including this file only on systems where we know it's missing.
  * Currently this is just some older versions of cygwin.
  */
-#if defined(HAVE_NETINET_IN_SYSTM_H) || !defined(__CYGWIN__)
+#if 0
 # include <netinet/in_systm.h>
 #endif
 #include <netinet/in.h>
@@ -66,7 +66,7 @@
 
 /* Is IPv6 supported by the library? */
 
-#if defined(AF_INET6) && defined(IN6ADDR_LOOPBACK_INIT) \
+#if 0 && defined(AF_INET6) && defined(IN6ADDR_LOOPBACK_INIT) \
 	&& defined(HAVE_INET_NTOP) && defined(HAVE_INET_PTON)
 # define SUPPORT_IPV6 1
 #endif
diff -r --unified zsh-5.9a/Src/Modules/tcp.mdd zsh-5.9/Src/Modules/tcp.mdd
--- zsh-5.9a/Src/Modules/tcp.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/tcp.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/net/tcp
-link=dynamic
+link=no
 load=no
 functions='Functions/TCP/*'
 
diff -r --unified zsh-5.9a/Src/Modules/watch.mdd zsh-5.9/Src/Modules/watch.mdd
--- zsh-5.9a/Src/Modules/watch.mdd	2022-05-08 06:18:22.000000000 +0000
+++ zsh-5.9/Src/Modules/watch.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/watch
-link=dynamic
+link=static
 load=yes
 
 autofeatures="b:log p:WATCH p:watch"
diff -r --unified zsh-5.9a/Src/Modules/zftp.mdd zsh-5.9/Src/Modules/zftp.mdd
--- zsh-5.9a/Src/Modules/zftp.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/zftp.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/zftp
-link=dynamic
+link=no
 load=no
 functions='Functions/Zftp/*'
 
diff -r --unified zsh-5.9a/Src/Modules/zprof.mdd zsh-5.9/Src/Modules/zprof.mdd
--- zsh-5.9a/Src/Modules/zprof.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/zprof.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/zprof
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:zprof"
diff -r --unified zsh-5.9a/Src/Modules/zpty.mdd zsh-5.9/Src/Modules/zpty.mdd
--- zsh-5.9a/Src/Modules/zpty.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/zpty.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/zpty
-link=dynamic
+link=static
 load=no
 
 autofeatures="b:zpty"
diff -r --unified zsh-5.9a/Src/Modules/zselect.mdd zsh-5.9/Src/Modules/zselect.mdd
--- zsh-5.9a/Src/Modules/zselect.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/zselect.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/zselect
-link=dynamic
+link=static
 load=no
 
 objects="zselect.o"
diff -r --unified zsh-5.9a/Src/Modules/zutil.mdd zsh-5.9/Src/Modules/zutil.mdd
--- zsh-5.9a/Src/Modules/zutil.mdd	2017-12-04 14:09:36.000000000 +0000
+++ zsh-5.9/Src/Modules/zutil.mdd	2023-12-05 00:00:29.937471863 +0000
@@ -1,5 +1,5 @@
 name=zsh/zutil
-link=either
+link=static
 load=yes
 
 moddeps="zsh/complete"
diff -r --unified zsh-5.9a/Src/exec.c zsh-5.9/Src/exec.c
--- zsh-5.9a/Src/exec.c	2022-05-08 06:18:22.000000000 +0000
+++ zsh-5.9/Src/exec.c	2023-12-05 00:00:29.941472304 +0000
@@ -267,7 +267,7 @@
 /* the resource limits for the shell and its children */
 
 /**/
-mod_export struct rlimit current_limits[RLIM_NLIMITS], limits[RLIM_NLIMITS];
+mod_export struct rlimit current_limits[16], limits[16];
 
 /**/
 mod_export int
diff -r --unified zsh-5.9a/aczsh.m4 zsh-5.9/aczsh.m4
--- zsh-5.9a/aczsh.m4	2022-05-08 06:18:22.000000000 +0000
+++ zsh-5.9/aczsh.m4	2023-12-05 00:00:29.941472304 +0000
@@ -418,7 +418,9 @@
     AC_TRY_COMMAND($CC -c $CFLAGS $CPPFLAGS $DLCFLAGS conftest1.c 1>&AS_MESSAGE_LOG_FD) &&
     AC_TRY_COMMAND($DLLD -o conftest1.$DL_EXT $LDFLAGS $DLLDFLAGS conftest1.o $LIBS 1>&AS_MESSAGE_LOG_FD); then
     save_ldflags=$LDFLAGS
-    LDFLAGS="$LDFLAGS $EXTRA_LDFLAGS -s"
+    dnl -s is removed here to prevent stripping executables,
+    dnl making this test a no-op
+    LDFLAGS="$LDFLAGS $EXTRA_LDFLAGS"
     AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdlib.h>
 #ifdef HPUX10DYNAMIC
diff -r --unified zsh-5.9a/configure.ac zsh-5.9/configure.ac
--- zsh-5.9a/configure.ac	2022-05-08 06:18:22.000000000 +0000
+++ zsh-5.9/configure.ac	2023-12-05 00:00:29.941472304 +0000
@@ -513,21 +513,21 @@
      s/ *$//'
 
 case " $LDFLAGS " in
-  *" -s "*) strip_exeldflags=true strip_libldflags=true
+  *" -s "*) strip_exeldflags=false strip_libldflags=false
     LDFLAGS=`echo " $LDFLAGS " | sed "$sed"` ;;
   *) strip_exeldflags=false strip_libldflags=false ;;
 esac
 
 case " ${EXELDFLAGS+$EXELDFLAGS }" in
   " ") ;;
-  *" -s "*) strip_exeldflags=true
+  *" -s "*) strip_exeldflags=false
     EXELDFLAGS=`echo " $EXELDFLAGS " | sed "$sed"` ;;
   *) strip_exeldflags=false ;;
 esac
 
 case " ${LIBLDFLAGS+$LIBLDFLAGS }" in
   " ") ;;
-  *" -s "*) strip_libldflags=true
+  *" -s "*) strip_libldflags=false
     LIBLDFLAGS=`echo " $LIBLDFLAGS " | sed "$sed"` ;;
   *) strip_libldflags=false ;;
 esac
--- zsh-5.9/Src/Modules/regex.mdd	2017-12-04 14:09:36.000000000 +0000
+++ -	2024-01-10 02:59:55.818926888 +0000
@@ -1,8 +1,5 @@
 name=zsh/regex
-link=`if test x$ac_cv_func_regcomp = xyes && \
-         test x$ac_cv_func_regexec = xyes && \
-         test x$ac_cv_func_regerror = xyes && \
-         test x$ac_cv_func_regfree = xyes; then echo dynamic; else echo no; fi`
-load=no
+link=static
+load=yes
 
 objects="regex.o"
--- zsh-5.9/Src/Modules/mapfile.c	2018-06-24 00:14:58.000000000 +0000
+++ -	2024-06-24 06:05:10.536447780 +0000
@@ -39,7 +39,7 @@
  * I don't know what I'm doing.
  */
 #if defined(HAVE_SYS_MMAN_H) && defined(HAVE_FTRUNCATE)
-#if defined(HAVE_MMAP) && defined(HAVE_MUNMAP) && defined(HAVE_MSYNC)
+#if 0 && defined(HAVE_MMAP) && defined(HAVE_MUNMAP) && defined(HAVE_MSYNC)
 #define USE_MMAP 1
 
 #include <sys/mman.h>
