diff --git a/berry/Makefile b/berry/Makefile
index a84af6b..cbb5303 100755
--- a/berry/Makefile
+++ b/berry/Makefile
@@ -1,12 +1,12 @@
-CFLAGS      = -Wall -Wextra -std=c99 -pedantic-errors -O2
+CFLAGS      = -Wall -Wextra -std=c99 -pedantic-errors -O2 -I$(COSMOS)/include
 DEBUG_FLAGS = -O0 -g -DBE_DEBUG
 TEST_FLAGS  = $(DEBUG_FLAGS) --coverage -fno-omit-frame-pointer -fsanitize=address -fsanitize=undefined
-LIBS        = -lm
+LIBS        = -static -lm -ldl -lreadline -lncurses -ltinfo
 TARGET      = berry
-CC          = gcc
+CC          = $(COSMO)/cosmocc/bin/$(ARCH)-unknown-cosmo-cc
 MKDIR       = mkdir
-LFLAGS      =
-PREFIX      = /usr/local
+LFLAGS      = -L$(COSMOS)/lib
+PREFIX      = $(COSMOS)
 BINDIR      = $(PREFIX)/bin
 
 INCPATH     = src default
diff --git a/berry/testall.be b/berry/testall.be
index 57414db..7a86829 100755
--- a/berry/testall.be
+++ b/berry/testall.be
@@ -1,7 +1,7 @@
 #! ./berry
 import os
 
-os.system('lcov', '-q -c -i -d . -o init.info')
+# os.system('lcov', '-q -c -i -d . -o init.info')
 
 var exec = './berry'
 var path = 'tests'
@@ -38,6 +38,7 @@ var cmds = [
 ]
 
 for cmd : cmds
+    break
     if os.system(cmd)
         os.exit(-1)
     end
--- -	2023-11-04 02:24:52.727430851 +0000
+++ b/berry/default/berry_conf.h	2023-10-29 05:32:39.828938709 +0000
@@ -162,7 +162,7 @@
  * otherwise disable the feature.
  * Default: 1
  **/
-#define BE_USE_SHARED_LIB               1
+#define BE_USE_SHARED_LIB               0
 
 /* Macro: BE_USE_OVERLOAD_HASH
  * Allows instances to overload hash methods for use in the
