diff -r --unified readline-8.2a/input.c readline-8.2/input.c
--- readline-8.2a/input.c	2022-04-08 19:43:24.000000000 +0000
+++ readline-8.2/input.c	2023-12-04 23:51:34.830635542 +0000
@@ -382,6 +382,7 @@
 
   tty = fileno (rl_instream);
 
+#if 0
 #if defined (HAVE_PSELECT) || defined (HAVE_SELECT)
   FD_ZERO (&readfds);
   FD_ZERO (&exceptfds);
@@ -406,7 +407,7 @@
   if (isatty (tty))
     return (_kbhit ());
 #endif
-
+#endif
   return 0;
 }
 
--- readline-8.2a/readline.pc.in	2019-04-09 02:22:47.000000000 +0800
+++ readline-8.2/readline.pc.in	2019-04-09 02:22:47.000000000 +0800
@@ -7,6 +7,5 @@
 Description: Gnu Readline library for command line editing
 URL: http://tiswww.cwru.edu/php/chet/readline/rltop.html
 Version: @LIBVERSION@
-Requires.private: @TERMCAP_PKG_CONFIG_LIB@
-Libs: -L${libdir} -lreadline
+Libs: -L${libdir} -lreadline -l@TERMCAP_PKG_CONFIG_LIB@
 Cflags: -I${includedir}
